File: java/javadpbo/src/com/sem4dpbo/week2/praktikum2/latihanpraktikum2_java/MainProduct.java
package com.sem4dpbo.week2.praktikum2.latihanpraktikum2_java;

import java.util.Scanner;

public class MainProduct {
    public static void main(String[] args) {
        //input masukan
        Scanner scan = new Scanner(System.in);

        //variabel-variabel
        int i;
        int n = 0;

        //masukan jumlah data
        try {
            System.out.print("Masukan jumlah data : ");
            n = scan.nextInt();
        }catch (Exception e){}

        Memory[] dataProduct3 = new Memory[n];          //deklarasi obcet dari class Memory

//        masukan data
        String idProduct = null, brand = null, model = null;
        int frequency = 0, memorySize = 0;
        long price = 0;
        boolean supportCuda = false;

        for (i = 0; i < n; i++) {
            try {
                //masukan data
                System.out.println("Masukan id dan harga product ke-"+(i+1)+":");
                idProduct = scan.next();
                price = scan.nextLong();
                System.out.println("Masukan brand dan model product ke-"+(i+1)+":");
                brand = scan.next();
                model = scan.next();
                System.out.println("Masukan memory frequency, memory size dan Cuda(true/false) product ke-"+(i+1)+":");
                frequency = scan.nextInt();
                memorySize = scan.nextInt();
                supportCuda = scan.nextBoolean();

            }catch (Exception e){}
                //memasukan data ke dalam object
            dataProduct3[i] = new Memory();
            dataProduct3[i].setIdProduct(idProduct);
            dataProduct3[i].setPrice(price);
            dataProduct3[i].setBrand(brand);
            dataProduct3[i].setModel(model);
            dataProduct3[i].setFrequency(frequency);
            dataProduct3[i].setMemorySize(memorySize);
            dataProduct3[i].setSupportsCuda(supportCuda);
            System.out.println("------------------------------------------");
        }

        dataProduct3[0].setIdProduct("111");
//        print data pada object
        for (i = 0; i < n; i++) {
            System.out.println("=============================================");
            System.out.println("Id Product   : "+dataProduct3[i].getIdProduct());
            System.out.println("Price        : "+dataProduct3[i].getPrice());
            System.out.println("Brand        : "+dataProduct3[i].getBrand());
            System.out.println("Model        : "+dataProduct3[i].getModel());
            System.out.println("Frequency    : "+dataProduct3[i].getFrequency());
            System.out.println("Memory Size  : "+dataProduct3[i].getMemorySize());
            System.out.println("Support Cuda : "+dataProduct3[i].isSupportsCuda());
        }
    }
}

File: cpp/2nd_Week/Praktikum2/latihanpraktikum2_cpp/MainProduct.cpp
#include "Memory.cpp"

int main()
{
    Memory data;        //instansiasi class memory dengan objcet data

    //memasukan data ke dalam object data
    data.setIdProduct("101");
    data.setPrice(15000000);
    data.setBrand("Lenovo");
    data.setModel("S340");
    data.setFrequency("3200 MHZ");
    data.setMemorySize("8 GB");
    data.setSupportsCuda("false");

    // print data
    cout << "======================" << endl;
    cout << "=    Data Product    =" << endl;
    cout << "======================" << endl;
    cout << "1. Id Product   : " << data.getIdProduct() << endl;
    cout << "2. Price        : " << data.getPrice() << endl;
    cout << "3. Brand        : " << data.getBrand() << endl;
    cout << "4. Model        : " << data.getModel() << endl;
    cout << "5. Freqeuency   : " << data.getFrequency() << endl;
    cout << "6. Memory Size  : " << data.getMemorySize() << endl;
    cout << "7. Supports Cuda: " << data.isSupportsCuda() << endl;
    
    return 0;
}
